fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/GroupByTest.java

```diff
@@ -237,33 +253,234 @@ private Reference constructGroupByPlan(final boolean withPredicateInSelectWhere,
+            }
+        }
+        assertEquals(expectedResult, bitMapGroupByStrValue);
+    }
+
+    @DualPlannerTest(planner = DualPlannerTest.Planner.CASCADES)
+    void testBitMapWithBitMapIndex() {
+        int bitBucketSize = 4;
+        RecordMetaDataHook hook = setupHookAndAddData(true, true, true, bitBucketSize);
+
+        final var cascadesPlanner = (CascadesPlanner)planner;
+        final var plan = cascadesPlanner.planGraph(
+                () -> constructBitMapGroupByPlan(bitBucketSize, true),
+                Optional.empty(),
+                IndexQueryabilityFilter.TRUE,
+                EvaluationContext.empty()).getPlan();
+
+        final Map<Pair<String, Integer>, List<Long>> expectedResult = new HashMap<>();
+        expectedResult.put(Pair.of("1", 0), List.of(2L)); // 1
+        expectedResult.put(Pair.of("1", 4), List.of(8L)); // 7
+        expectedResult.put(Pair.of("2", 0), List.of(2L)); // 1
+        expectedResult.put(Pair.of("3", 0), List.of(2L)); // 1
+        expectedResult.put(Pair.of("4", 0), List.of(4L)); // 2
+        expectedResult.put(Pair.of("4", 4), List.of(2L)); // 5
+        expectedResult.put(Pair.of("5", 0), List.of(4L)); // 2
+        expectedResult.put(Pair.of("6", 0), List.of(4L)); // 2
+        expectedResult.put(Pair.of("7", 0), List.of(8L)); // 3
+        expectedResult.put(Pair.of("7", 4), List.of(4L)); // 6
+        expectedResult.put(Pair.of("8", 0), List.of(8L)); // 3
+        expectedResult.put(Pair.of("9", 0), List.of(8L)); // 3
+        expectedResult.put(Pair.of("10", 4), List.of(1L)); // 4
+        expectedResult.put(Pair.of("11", 4), List.of(1L)); // 4
+
+        final Map<Pair<String, Integer>, List<Long>> bitMapGroupByStrValue = new HashMap<>();
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, hook);
+            try (RecordCursor<QueryResult> cursor = FDBSimpleQueryGraphTest.executeCascades(recordStore, plan, Bindings.EMPTY_BINDINGS)) {
+                cursor.forEach(queryResult -> {
+                    final Message queriedMessage = queryResult.getMessage();
+                    final Descriptors.Descriptor recDescriptor = queriedMessage.getDescriptorForType();
+                    String strValue = (String) queriedMessage.getField(recDescriptor.findFieldByName("str_value_indexed"));
+                    Integer bitBucketValue = (int) queriedMessage.getField(recDescriptor.findFieldByName("bit_bucket_num_value2"));
+                    byte[] value = ((ByteString) queriedMessage.getField(recDescriptor.findFieldByName("bitmap_field"))).toByteArray();
+                    bitMapGroupByStrValue.put(Pair.of(strValue, bitBucketValue), AggregateValueTest.collectOnBits(value));
+                }).join();
+            }
+        }
+        assertEquals(expectedResult, bitMapGroupByStrValue);
+    }
+
+    @Nonnull
+    private Reference constructBitMapGroupByPlan(int bucketSize, boolean includeBitBucketFunc) {
```

Comment:
![IMG_B4D3C3C7B7AC-1 2](https://github.com/user-attachments/assets/aa77fb29-e0f3-4181-8a77-de0a41dc5502)

I had a quick look at the plan and it does not look correct to me, namely:
1. The grouping keys group in `select-where` should be gone, in https://github.com/FoundationDB/fdb-record-layer/pull/2762 I introduced a new simplified version of aggregate plans that removes the grouping key from the `select-where`, it moves it up to the `group-by` expression instead.
2. The grouping keys does not include the `bit_bucket` function.

Reply:
discussed offline and modified the code.